loom/common/assets/assetProtocol.cpp

```diff
@@ -198,9 +198,13 @@ bool AssetProtocolHandler::readFrame()
     // Allocate a buffer to store data.
     bytes = (unsigned char *)lmAlloc(NULL, frameLength);
 
-    // Make sure we can peek that much data.
     bytesLength = frameLength;
-    loom_net_readTCPSocket(socket, bytes, &bytesLength, 1);
+
+    // Wait until bytesLength bytes are ready for reading
+    // TODO This could probably be implemented better, but
+    // it doesn't break with larger frames like the previous
+    // implementation did at least.
+    loom_net_readTCPSocket(socket, bytes, &bytesLength, 0);
```

Comment:
Yeah I think that was exactly it. And the OS recv buffer usually isn't that big, so it's easy to reach with even moderate amounts of data (like a page of telemetry JSON).